lib/labelhash.cc

```diff
@@ -336,3 +346,176 @@ LabelHash::~LabelHash()
+        unsigned char version, ht_type;
+
+        infile.read((char *) &version, 1);
+        infile.read((char *) &ht_type, 1);
+        if (!(version == SAVED_FORMAT_VERSION)) {
+            std::ostringstream err;
+            err << "Incorrect file format version " << (int) version
+                << " while reading labels/tags from " << filename;
+            throw khmer_file_exception(err.str());
+        } else if (!(ht_type == SAVED_LABELSET)) {
+            std::ostringstream err;
+            err << "Incorrect file format type " << (int) ht_type
+                << " while reading labels/tags from " << filename;
+            throw khmer_file_exception(err.str());
+        }
+
+        infile.read((char *) &save_ksize, sizeof(save_ksize));
+        if (!(save_ksize == graph->ksize())) {
+            std::ostringstream err;
+            err << "Incorrect k-mer size " << save_ksize
+                << " while reading labels/tags from " << filename;
+            throw khmer_file_exception(err.str());
+        }
+    } catch (std::ifstream::failure &e) {
+        std::string err;
+        err = "Unknown error reading header info from: " + filename;
+        throw khmer_file_exception(err);
+    }
+
+    char * buf = new char[IO_BUF_SIZE];
+
+    unsigned int loaded = 0;
+    long remainder;
+
+
+    HashIntoType * kmer_p = NULL;
+    Label * labelp = NULL;
+
+    remainder = 0;
+    unsigned int iteration = 0;
+    while (!infile.eof()) {
+      //std::cout << "XXX " << loaded << " : " << "\n";
+        unsigned int i;
+
+        try {
+            infile.read(buf + remainder, IO_BUF_SIZE - remainder);
+        } catch (std::ifstream::failure &e) {
+
+            // We may get an exception here if we fail to read all the
+            // expected bytes due to EOF -- only pass it up if we read
+            // _nothing_.  Note that the while loop exits on EOF.
+
+            if (infile.gcount() == 0) {
+                std::string err;
+                err = "Unknown error reading data from: " + filename;
+                //std::cout << err;
+                throw khmer_file_exception(err);
+            }
+        }
+
+        long n_bytes = infile.gcount() + remainder;
+        remainder = n_bytes % (sizeof(Label) + sizeof(HashIntoType));
+        n_bytes -= remainder;
+
+        iteration++;
+
+        for (i = 0; i < n_bytes;) {
+            kmer_p = (HashIntoType *) (buf + i);
+            i += sizeof(HashIntoType);
+
+            labelp = (Label *) (buf + i);
+            i += sizeof(Label);
+
+            Label * labelp2;
+
+            // std::cout << *kmer_p << " - " << *labelp << "\n";
+
+            graph->all_tags.insert(*kmer_p);
+            labelp2 = check_and_allocate_label(*labelp);
+            link_tag_and_label(*kmer_p, *labelp2);
+
+            loaded++;
+        }
+        if (!(i == n_bytes)) {
+          //std::cout << "XYZ " << i << " : " << n_bytes << "\n";
+          throw khmer_exception();
```

Comment:
This branch has no code coverage. Also, please pass a description when throwing `khmer_exception`